integration/e2e_test.go

```diff
@@ -90,9 +99,14 @@ func TestSnapshotCacheSingleEnvoyAndXdsRelayServer(t *testing.T) {
 				}, 1*time.Second, 100*time.Millisecond).Should(gomega.Equal(nListeners))
 			}
 
+			// If envoy never starts, the buffer is empty
+			assert.NotEmpty(t, envoyLogsBuffer.String())
 			// TODO(https://github.com/envoyproxy/xds-relay/issues/66): figure out a way to only only copy
 			// envoy logs in case of failures.
 			testLogger.With("envoy_logs", envoyLogsBuffer.String()).Debug(ctx, "captured envoy logs")
+
+			err := stopEnvoy(ctx, pid)
```

Comment:
Should this be a separate test? I think we want to test the codepath where envoy is receiving continuous updates without restarting.

Reply:
We have to start a new envoy instance to allow a new bootstrap. We can use separate port numbers to start a new instance of envoy, or shutdown the old one and start a new one. I like shutting and restarting more since its clean and will surface more bugs.

Reply:
@jessicayuen , if we want to test the scenario where the control plane sends v2 and v3 interspersed we should create a different test that tests this specifically.

@jyotimahapatra , now that we're killing envoy manually, can you try removing the Linux-only check I added and check if the tests fail? You can inspect if there's any dangling process after tests run by inspecting the docker container (run `docker run --rm -it --pid host jess/htop` in a separate window and search for the envoy process)

Reply:
I verified that the verification steps mentioned work. Also, i think we have safety due to these:
1. If kill gets stuck for any reason, the test will timeout .
2. If kill gives back an error, the test fails due to err check.